vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_client.go

```diff
@@ -114,14 +114,13 @@ func newContainerMetricsSnapshot(containerMetrics v1beta1.ContainerMetrics, podM
 }
 
 func calculateUsage(containerUsage k8sapiv1.ResourceList) model.Resources {
-	cpuQuantity := containerUsage[k8sapiv1.ResourceCPU]
-	cpuMillicores := cpuQuantity.MilliValue()
-
-	memoryQuantity := containerUsage[k8sapiv1.ResourceMemory]
-	memoryBytes := memoryQuantity.Value()
-
-	return model.Resources{
-		model.ResourceCPU:    model.ResourceAmount(cpuMillicores),
-		model.ResourceMemory: model.ResourceAmount(memoryBytes),
+	result := map[model.ResourceName]model.ResourceAmount{}
+	for resourceName, quantity := range containerUsage {
```

Comment:
We're going to drop all samples that aren't CPU or memory later and log an error (for example [here](https://github.com/kubernetes/autoscaler/blob/d17b1aaaec7fdb2c2afad18ce8c076f01e0f6f4b/vertical-pod-autoscaler/pkg/recommender/model/types.go#L94)), VPA doesn't provide recommendation for other resources. Please return only CPU or memory samples.

Reply:
In our kubernetes clusters are now running a custom recommender that could have other resources than CPU and Mem.
At least admission and updater should be completely generic in terms of resource Name and Count.
Should we try to do the same in the recommender? I mean handle as much as possible a list of resource instead of a pair CPU + Mem ?

Reply:
If you want to change recommender code to make it easier to reuse in custom recommenders I think it's a good idea.

But we need to do that in a way that doesn't cause a bunch of spurious error logs in the default recommender

Reply:
It seems previously this function return both CPU and memory. After this change, it might only return one of the resource or none of them. Is there any downstream consumers of this function that explicitly expect both CPU and memory to be present and will panic or have unexpected behaviour when only one or none of the resource is returned?